sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -129,24 +130,25 @@ object InternalRow {
    */
   def getAccessor(dt: DataType, nullable: Boolean = true): (SpecializedGetters, Int) => Any = {
     val getValueNullSafe: (SpecializedGetters, Int) => Any = dt match {
```

Comment:
instead of
```
dt match {
  case _ => dt.physicalDataType match ...
}
```
we can just do
```
dt.physicalDataType match ...
```

Reply:
Makes sense. We have to introduce a `PhysicalUserDefinedType` to remove this layer of indirection, but this is very doable.

Reply:
Ah, if we can remove `PhysicalUserDefinedType`, it's OK to make this pattern match a bit longer.

Reply:
Left this here as discussed. This additional pattern match also allows us to remove `PhysicalObjectType` which is nice.